tests/harnesses/ld-floxlib/test-load-library-last.sh

```diff
@@ -0,0 +1,54 @@
+### Verify environment
+set -ex
+
+# FLOX_ENV_LIB_DIRS is defined
+test -n "$FLOX_ENV_LIB_DIRS"
+
+# LD_AUDIT is defined, exists and points to ld-floxlib.so
+test -e "$LD_AUDIT"
+[[ "$LD_AUDIT" == */ld-floxlib.so ]]
+
+# LD_LIBRARY_PATH is not defined
+test -z "$LD_LIBRARY_PATH"
+
+### Test 1: load libraries found in $FLOX_ENV_LIBS last
+
+# Build glibc version probe using "system" version of glibc.
+cc -o get-glibc-version ./get-glibc-version.c
+
+# Remove all the tricks that Nix does to force this executable
+# to run with the exact version it was compiled with.
+
+# 1. Remove the custom RUNPATH added by gcc-wrapper.
+patchelf --remove-rpath ./get-glibc-version
+
+# 2. set the interpreter to the one used by the default system.
```

Comment:
question nonblocking: what does the interpreter do, and how does it affect the test?

Reply:
The dynamic linker interpreter is the entrypoint for kicking off the process of resolving dynamic library references - think of it like a "shebang" for ELF binaries, just like "#!/usr/bin/env" or something like that. Notably the Nix version of this interpreter is hardcoded to always look in the `lib` subdirectory of the specific glibc package within which it resides, e.g. note how these two binaries on a NixOS system are built to look in different glibc packages by default:
```
[nix-shell:~]$ patchelf --print-interpreter `which patchelf`
/nix/store/yzjgl0h6a3qh1mby405428f16xww37h0-glibc-2.35-224/lib/ld-linux-x86-64.so.2

[nix-shell:~]$ patchelf --print-interpreter /run/current-system/sw/bin/ls
/nix/store/9y8pmvk8gdwwznmkzxa6pwyah52xy3nk-glibc-2.38-27/lib/ld-linux-x86-64.so.2
```

To defeat this "failsafe" behaviour (and simulate what you'll find on any other Linux system) we need to replace the interpreter with one that looks in a different place, and we can be pretty certain that the interpreter used with `/bin/sh` is a) the one [1] that corresponds to the system itself and b) will be different from the "OLDER" glibc version that we're pinning for use by this test.  Hope that makes sense!

Reply:
Definitely thanks for the in-depth explanation!